Step5 - 로또(수동)

Sung-jin opened the pull request:
로또 수동 기능 구현 리뷰 부탁드립니다.

모든 객체를 래핑 객체로 변경도 같이 진행했는데, 생각보다 많은 곳이 변경이 되었네요.
혹시 리뷰 하기 힘드시다면, 기능 구현과 리펙토링을 별도로 리뷰 요청하겠습니다.

Sung-jin committed:
test: 1단계 - String 클래스에 대한 학습 테스트

Sung-jin committed:
test: 1단계 - Set Collection 에 대한 학습 테스트

Sung-jin committed:
Merge branch 'next-step:sung-jin' into sung-jin

Sung-jin committed:
Merge branch 'sung-jin' of https://github.com/next-step/java-lotto-pro into sung-jin_(origin)

Sung-jin committed:
Merge branch 'sung-jin' of https://github.com/next-step/java-lotto-pro into sung-jin_(origin)

Sung-jin committed:
docs: 로또 수동 기능 요구사항 정리

Sung-jin committed:
feat: 로또 구입 시 수동으로 입력하는 기능 구현

Sung-jin committed:
feat: 모든 입력 예외를 처리하여 입력 받도록 수정

Sung-jin committed:
test: Lotto 객체의 당첨 비교 시 인자값 타입 변경에 따른 유효성 검사 책임 변경 대응

Sung-jin committed:
refactor: 양수 객체 문자열로 생성 시 null/empty 일 때 0 으로 컨버팅 하는 옵션 추가

Sung-jin committed:
refactor: money 객체의 값을 래핑 클래스로 변경 및 money 를 사용하던 기존 로직들 money 로 래핑

Sung-jin committed:
refactor: LottoNumber 의 값을 PositiveNumber 로 변경 및 관련된 로직 변경 대응

Sung-jin committed:
refactor: 로또 당첨 순위의 정답 수를 PositiveNumber 로 매핑 및 관련 로직 대응

Sung-jin committed:
fix: 문자열로 로또 번호 리스트를 생성할 때 띄어쓰기 없이 콤마로 구분하면 생성 실패하는 현상 수정

Sung-jin committed:
refactor: Lotto 객체의 인자값들을 래핑 객체로 변경

Sung-jin committed:
fix: 양수 래핑 클래스 생성자 타입을 int -> long 으로 변경

seok-2-o commented on src/main/java/camp/nextstep/edu/common/PositiveNumber.java:
```diff
@@ -1,28 +1,60 @@
+    private static final long MIN_VALUE = 0L;
 
-    public final Integer value;
+    private final long value;
```
👍

seok-2-o commented on src/main/java/camp/nextstep/edu/level1/lotto/lotto/LottoNumber.java:
```diff
@@ -1,44 +1,30 @@
     private static final int LOTTO_END_NUMBER = 45;
 
-    private final int number;
+    private final PositiveNumber number;
```
LottoNumber 는 PositiveNumber에 포함되긴하지만 훨씬 작은 범위를 표한하는 숫자인데
PositiveNumber 를 구현할 필요가 있을까요? PositiveNumber 를 구현했을떄 장점은 무엇인가요?

Sung-jin replied:
생각해보니, LottoNumber 자체가 이미 래핑 클래스네요.
모든 값을 다 래핑을 해야겠다라고만 생각해서 리펙토링 해야겠다라고 생각했던 것 같습니다.

말씀하신 것처럼 이전에 LottoNumber 자체가 이미 래핑 클래스로 구현하였고 validation 까지 동작했었으니, 해당 부분은 롤백하겠습니다.

seok-2-o commented on src/main/java/camp/nextstep/edu/level1/lotto/lotto/LottoNumber.java:
```diff
@@ -1,44 +1,30 @@
-    private void checkValidLottoNumber(int number) {
-        if (number < LOTTO_START_NUMBER || number > LOTTO_END_NUMBER) {
+    private void checkValidLottoNumber(PositiveNumber number) {
+        if (number.getValue() < LOTTO_START_NUMBER || number.getValue() > LOTTO_END_NUMBER) {
```
PositiveNumber 를 구현했기때문에 오히려 getter 를 사용하게 된 것 같아요.

Sung-jin replied:
넵. 해당 부분도 위와 같은 이유로 다시 롤백하겠습니다.

seok-2-o reviewed (changes requested):
마지막 미션도 큰 구조 변화 없이 잘 진행해주셨네요.
다만 PositiveNumber를 활용하신 부분에 대해서는 의문이 있어 코멘트 남겨놓았어요

수정해야하는 부분은 아니니 답변해주시면 좋을 것 같아요.

Sung-jin committed:
refactor: 리뷰 적용

기존 LottoNumber 의 멤버 변수를 PositiveNumber -> 기본 primitve 타입으로 변경
LottoNumber 클래스 가체가 이미 래핑 클래스임.

seok-2-o reviewed (commented):
-

seok-2-o reviewed (approved):
피드백 빠르게 반영잘해주셨네요 💯
첫번쨰 미션 하느라 고생 많으셨습니다 :)

이번 피드백을 마지막으로 저와의 미션은 마무리가 되네요 🙇

저와 미션은 여기까지지만 앞으로 궁금한점이나 어려운점 있으면 편하게 DM 주세요 :)
좋은 개발자로 성장하는데 조금이나마 도움이 되기를 바라봅니다.

감사드리며,
이재석 드림

Sung-jin commented:
저도 이번 한주간 정말 많은 것을 얻었습니다.
리뷰어님도 많은 것을 얻는 기회가 되었으면 좋을 것 같습니다.

첫번째 미션을 진행을 해주셔서 감사합니다.
